Gems/ROS2/Code/Source/Camera/CameraSensor.cpp

```diff
@@ -208,6 +208,7 @@ namespace ROS2
                 message.step = message.width * Internal::BitDepth.at(format);
                 message.data = std::vector<uint8_t>(result.m_dataBuffer->data(), result.m_dataBuffer->data() + result.m_dataBuffer->size());
                 message.header = header;
+                postProcess(message);
```

Comment:
I would see the `postProcess` function that takes reference `result` and fill the message here.
Justification: when I use OpenCV I can create cvMat around buffer `result.m_dataBuffer` and I potentially have one copy less.

Reply:
The design is changed, please review the new one